src/accelerate/utils/parallel_state.py

```diff
@@ -0,0 +1,790 @@
+# Copyright (c) 2022, NVIDIA CORPORATION. All rights reserved.
```

Comment:
```suggestion
# Copyright (c) 2022, NVIDIA CORPORATION. All rights reserved.
# Adapted from https://github.com/NVIDIA/Megatron-LM/blob/main/megatron/core/parallel_state.py
```
probably a good idea to mentioned the source - I didn't check but it looks like this is where you took it from.
```